modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -226,10 +229,33 @@ const FrontendDataSet = ({
 	const isMounted = useIsMounted();
 
 	function updateDataSetItems(dataSetData) {
-		setTotal(dataSetData.totalCount);
+		setCollectionActions(dataSetData.actions);
 		setItems(dataSetData.items);
+		setTotal(dataSetData.totalCount);
 	}
 
+	useEffect(() => {
+		const filteredCreationMenu = {};
+
+		if (creationMenu && collectionActions) {
+			if (creationMenu.primaryItems) {
+				filteredCreationMenu.primaryItems = filterCreationActions(
+					creationMenu.primaryItems,
+					collectionActions
+				);
+			}
+
+			if (creationMenu.secondaryItems) {
+				filteredCreationMenu.secondaryItems = filterCreationActions(
```

Comment:
Check for `secondaryItems` because I found a reference in the [CreationMenu.js](https://github.com/liferay-frontend/liferay-portal/pull/3781/files#diff-24247200af872433ada68eede9358b593a7317efac577789ef187a8e42148d6cL134).
However, I haven't found any usage of it in the codebase

Reply:
I think you can safely burn with fire all logic related to `secondaryItems` :fire: 

It _is_ a building block in [CreationMenu backend object](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/util/CreationMenu.java), that we are passing into FDS in some usages, but FDS frontend implementation of CreationMenu ignores that property. If we add support for features like favorites in FDS, we can add support for secondary items as well.

Reply:
Great

Reply:
do we need an effect here? what if we compute `filteredCreationMenu` right inside `updateDataSetItems()`? we can even avoid the calculation if collectionActions don't change.